ext/rubymain.cpp

```diff
@@ -744,23 +744,30 @@ static VALUE t_set_sock_opt (VALUE self UNUSED, VALUE signature, VALUE lev, VALU
 	const void *v;
 	socklen_t len;
 
-	switch (TYPE(optval)) {
-	case T_FIXNUM:
-		i = FIX2INT(optval);
-		goto numval;
-	case T_FALSE:
-		i = 0;
-		goto numval;
-	case T_TRUE:
-		i = 1;
-		numval:
-		v = (void*)&i; len = sizeof(i);
-		break;
-	default:
-		StringValue(optval);
-		v = RSTRING_PTR(optval);
-		len = RSTRING_LENINT(optval);
-		break;
+	if (option == SO_BINDTODEVICE) {
```

Comment:
Since this is linux-only, I now thing we should wrap it in an `#ifdef SO_BINDTODEVICE`. For the consumer code, it looks like Ruby provides this constant since 1.8.7, so we're good for it being exposed:
http://rxr.whitequark.org/mri/source/ext/socket/socket.c?v=1.8.7-p374#4583